### ISSUE TITLE
Node crash due to Postgres 

### ISSUE THREAD

**Issue by engenegr** [ACINQ/eclair#1786]
Node failure due to `Cannot check database lease`. Multiple times, didn't notice this issue with default Postgres configs.

Maybe it would be good for node to not crash and just reject some operations with channels?

It repeats regularly, recently it lead to channel force-close.

Self-compiled. "0.5.1-98bb7be"

Linux Debian 10. OpenJDK 64-Bit Server VM (build 11.0.9.1+1-post-Debian-1deb10u2, mixed mode, sharing)

I had config
```
eclair.db.postgres.lease {
    interval = 60 seconds        // Default: 5 minutes
    renew-interval =  30 seconds // Default: 1 minute
}
```
I made lease times shorter for faster node restarts. What is suggested time intervals?

```
2021-05-05 16:36:41,615 ERROR f.a.eclair.db.Databases - fatal error: Cannot obtain lock on the database.
fr.acinq.eclair.db.pg.PgUtils$LockException: Cannot check database lease
	at fr.acinq.eclair.db.pg.PgUtils$.fr$acinq$eclair$db$pg$PgUtils$$checkDatabaseLease(PgUtils.scala:211)
	at fr.acinq.eclair.db.pg.PgUtils$LeaseLock.$anonfun$withLock$1(PgUtils.scala:94)
	at fr.acinq.eclair.db.pg.PgUtils$.inTransaction(PgUtils.scala:106)
	at fr.acinq.eclair.db.pg.PgUtils$.$anonfun$inTransaction$1(PgUtils.scala:121)
	at fr.acinq.eclair.db.jdbc.JdbcUtils.withConnection(JdbcUtils.scala:35)
	at fr.acinq.eclair.db.jdbc.JdbcUtils.withConnection$(JdbcUtils.scala:32)
	at fr.acinq.eclair.db.pg.PgUtils$.withConnection(PgUtils.scala:30)
	at fr.acinq.eclair.db.pg.PgUtils$.inTransaction(PgUtils.scala:120)
	at fr.acinq.eclair.db.pg.PgUtils$LeaseLock.withLock(PgUtils.scala:92)
	at fr.acinq.eclair.db.pg.PgChannelsDb.$anonfun$addOrUpdateChannel$1(PgChannelsDb.scala:53)
	at fr.acinq.eclair.db.Monitoring$Metrics$.$anonfun$withMetrics$1(Monitoring.scala:33)
	at fr.acinq.eclair.KamonExt$.time(PimpKamon.scala:28)
	at fr.acinq.eclair.db.Monitoring$Metrics$.withMetrics(Monitoring.scala:31)
	at fr.acinq.eclair.db.pg.PgChannelsDb.addOrUpdateChannel(PgChannelsDb.scala:53)
	at fr.acinq.eclair.channel.Channel$MyState.storing(Channel.scala:2434)
	at fr.acinq.eclair.channel.Channel$$anonfun$9.applyOrElse(Channel.scala:774)
	at fr.acinq.eclair.channel.Channel$$anonfun$9.applyOrElse(Channel.scala:656)
	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:35)
	at fr.acinq.eclair.channel.Channel$$anonfun$handleExceptions$1.applyOrElse(Channel.scala:2415)
	at fr.acinq.eclair.channel.Channel$$anonfun$handleExceptions$1.applyOrElse(Channel.scala:2412)
	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:35)
	at akka.actor.FSM.processEvent(FSM.scala:804)
	at akka.actor.FSM.processEvent$(FSM.scala:801)
	at fr.acinq.eclair.channel.Channel.processEvent(Channel.scala:103)
Caused by: org.postgresql.util.PSQLException: ERROR: could not serialize access due to read/write dependencies among transactions
  Detail: Reason code: Canceled on identification as a pivot, during conflict out checking.
  Hint: The transaction might succeed if retried.
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2533)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2268)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:313)
	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:448)
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:369)
	at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:310)
	at org.postgresql.jdbc.PgStatement.executeCachedSql(PgStatement.java:296)
	at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:273)
	at org.postgresql.jdbc.PgStatement.executeQuery(PgStatement.java:226)
	at com.zaxxer.hikari.pool.ProxyStatement.executeQuery(ProxyStatement.java:111)
	at com.zaxxer.hikari.pool.HikariProxyStatement.executeQuery(HikariProxyStatement.java)
	at fr.acinq.eclair.db.pg.PgUtils$.$anonfun$getCurrentLease$1(PgUtils.scala:221)
	at fr.acinq.eclair.db.jdbc.JdbcUtils.using(JdbcUtils.scala:50)
	at fr.acinq.eclair.db.jdbc.JdbcUtils.using$(JdbcUtils.scala:46)
	at fr.acinq.eclair.db.pg.PgUtils$.using(PgUtils.scala:30)
	at fr.acinq.eclair.db.pg.PgUtils$.getCurrentLease(PgUtils.scala:220)
	at fr.acinq.eclair.db.pg.PgUtils$.$anonfun$checkDatabaseLease$1(PgUtils.scala:197)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
	at scala.util.Try$.apply(Try.scala:210)
	at fr.acinq.eclair.db.pg.PgUtils$.fr$acinq$eclair$db$pg$PgUtils$$checkDatabaseLease(PgUtils.scala:197)
	at fr.acinq.eclair.db.pg.PgUtils$LeaseLock.$anonfun$withLock$1(PgUtils.scala:94)
	at fr.acinq.eclair.db.pg.PgUtils$.inTransaction(PgUtils.scala:106)
	at fr.acinq.eclair.db.pg.PgUtils$.$anonfun$inTransaction$1(PgUtils.scala:121)
	at fr.acinq.eclair.db.jdbc.JdbcUtils.withConnection(JdbcUtils.scala:35)
```

@rorp

**Comment by engenegr** [comment]
Looks like I can't fix it via config parameters.

https://stackoverflow.com/questions/21706858/org-postgresql-util-psqlexception-error-could-not-serialize-access-due-to-read

**Comment by engenegr** [comment]
Channel force-close seems irrelevant and happened 20 min earlier DB failure. 
With error 
```
peer sent error: ascii='remote error' bin=72656d6f7465206572726f72
```

**Comment by pm47** [comment]
How long have you been running postgres before encountering this issue?

We have been reworking the postgres code lately, and this has most certainly been fixed by #1762. Feel free to compile and run latest master.

As for the lease interval, the suggested time intervals are the default values. You can tweak them a little bit, but I wouldn't go below 60s/30s.

**Comment by pm47** [comment]
> Channel force-close seems irrelevant and happened 20 min earlier DB failure.

Yeah, probably unrelated.

**Comment by engenegr** [comment]
@pm47 thank you for recommendations.

it was freshly installed postgres and it is running together with eclair. I had a prospect to use a plugin with Eclair so I compiled particular commit. I will look into running master as recommended.

**Comment by pm47** [comment]
We are moving to postgres on our main node too, so the postgres code will benefit from a much closer attention.

**Comment by engenegr** [comment]
@pm47 

caught this on "0.5.2-SNAPSHOT-90fbcd3"

```
2021-05-08 09:47:21,548 ERROR f.a.e.SimpleSupervisor - supervisor caught error for child=db-event-handler strategy=Resume org.postgresql.util.PSQLException: ERROR: could not serialize access due to concurrent update
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2533)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2268)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:313)
	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:448)
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:369)
	at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:159)
	at org.postgresql.jdbc.PgPreparedStatement.executeUpdate(PgPreparedStatement.java:125)
	at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeUpdate(ProxyPreparedStatement.java:61)
	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeUpdate(HikariProxyPreparedStatement.java)
	at fr.acinq.eclair.db.pg.PgChannelsDb.$anonfun$updateChannelMetaTimestampColumn$2(PgChannelsDb.scala:109)
	at fr.acinq.eclair.db.pg.PgChannelsDb.$anonfun$updateChannelMetaTimestampColumn$2$adapted(PgChannelsDb.scala:106)
	at fr.acinq.eclair.db.jdbc.JdbcUtils.using(JdbcUtils.scala:50)
	at fr.acinq.eclair.db.jdbc.JdbcUtils.using$(JdbcUtils.scala:46)
	at fr.acinq.eclair.db.pg.PgUtils$.using(PgUtils.scala:31)
	at fr.acinq.eclair.db.pg.PgChannelsDb.$anonfun$updateChannelMetaTimestampColumn$1(PgChannelsDb.scala:106)
	at fr.acinq.eclair.db.pg.PgChannelsDb.$anonfun$updateChannelMetaTimestampColumn$1$adapted(PgChannelsDb.scala:105)
	at fr.acinq.eclair.db.pg.PgUtils$.inTransaction(PgUtils.scala:251)
	at fr.acinq.eclair.db.pg.PgUtils$.$anonfun$inTransaction$1(PgUtils.scala:266)
	at fr.acinq.eclair.db.jdbc.JdbcUtils.withConnection(JdbcUtils.scala:35)
	at fr.acinq.eclair.db.jdbc.JdbcUtils.withConnection$(JdbcUtils.scala:32)
	at fr.acinq.eclair.db.pg.PgUtils$.withConnection(PgUtils.scala:31)
	at fr.acinq.eclair.db.pg.PgUtils$.inTransaction(PgUtils.scala:265)
	at fr.acinq.eclair.db.pg.PgChannelsDb.updateChannelMetaTimestampColumn(PgChannelsDb.scala:105)
	at fr.acinq.eclair.db.pg.PgChannelsDb.$anonfun$updateChannelMeta$1(PgChannelsDb.scala:123)
```

Node haven't crashed though.

**Comment by pm47** [comment]
This error is harmless but it's still a bug. Reproduced and fixed in e3a273b.

**Comment by engenegr** [comment]
Thank you @pm47
